design/block_explorer.md

```diff
@@ -8,65 +8,61 @@ that also displays private information belonging to that user.
-    * Latest transactions [TODO: CLARIFY THAT THIS IS JUST FOR THE CURRENT USER], with information on [TODO: I ASSUME THIS INFO IS DECRYPTED?]:
-      * Status
-      * From
-      * To
-      * State [TODO: CLARIFY WHAT THIS IS]
-      * Timestamp
-    * Number of transactions
-    * The bytecode and optionally sourcecode (via GitHub) of any deployed contracts [TODO: CLARIFY WHETHER THIS IS JUST FOR CURRENT USER]
-    * Ability to search for transactions by: [TODO: ADD SOME REQUIREMENT HERE AROUND SPEED AND NOT RESCANNING ENTIRE BLOCKCHAIN - I ASSUME THIS IS WHERE A LOT OF COMPLEXITY COMES IN?]
-      * Address
-      * Transaction hash
-      * Block
-      * Token [TODO: CLARIFY WHAT THIS MEANS]
-    * Network health (e.g. time to complete gossip)
-    * Revelation periods and associated transactions and contracts [TODO: CLARIFY WHAT THIS MEANS]
+  * The private block explorer has the same capabilities as the public block explorer. In addition, it allows the 
+    following:
+    * Displaying the decrypted transaction and transaction receipt if the user is allowed to view them
+    * Searching for transactions by:
+      * Address [TODO: CLARIFY WHAT THIS MEANS]
+      * Token symbol [TODO: CLARIFY WHAT THIS MEANS]
 
 ## Design
 
-TBD - Will depend on which alternative we select from the below.
+We will build our own private and public block explorers.
+
+Design is TBD.
 
 ## Known limitations
 
-TBD - Will depend on which alternative we select from the below.
+TBD
 
 ## Alternatives considered
 
 ### Build our own private and public block explorers
 
-TODO
+The downside of this approach is that we'd have to write all the block explorer logic from scratch. While displaying 
+individual rollups and transactions is relatively simple, the complexity emerges when processing the data to present a 
+useful view to the customer. For example, a user may wish to view all the transfers to their address by an ERC-20 
+contract, which requires walking the chain and storing the results locally so that they don't have to be continuously 
+recomputed.
 
-  * DOCUMENT WHAT'S EXPENSIVE FROM AN ENG PERSPECTIVE (DIGESTING CHAIN INTO DB, GETTING CONTRACTS?)
-  * TALK ABOUT NEED FOR DESIGN WORK, NO CURRENT CAPABILITY IN-HOUSE
-  * TALK ABOUT BENEFIT OF SHARED ARCH + LIBS FOR PUBLIC + PRIVATE EXPLORERS
-  * TALK ABOUT BENEFIT OF REUSING EXISTING STACK (GOLANG)
+However, a decisive upside of building our own block explorers is that Obscuro's rules about data visibility mean that 
+an off-the-shelf block explorer is unlikely to be fit for purpose in various ways, and will require extensive 
+customisation. We talk about that in section `Fork an existing block explorer for the public block explorer`, below.
 
-### Fork an existing block explorer for the public block explorer
+By writing our own block explorers, we can also reuse our existing stack (i.e. Go).
+
+### Fork an existing block explorer
```

Comment:
Given we've been unable to find any Go implementations of a block explorer, we're precluded from forking any existing explorer as we'd need to run some of the functionality through EGO

Reply:
Fixed.